src/main/java/technology/tabula/detectors/RegexSearch.java

```diff
@@ -0,0 +1,641 @@
+ *	    data data data data data	<-- Start of area to extract if(includeRegexBeforeTable==false)      | Subsection #1
+ *      data data data data data                                                                         | Subsection #1
+ *      data data data data data                                                                         | Subsection #1
+ *      ----------------------------------------(end of page)
+ *      data data data data data                                                                     | Subsection #2
+ *      data data data data data                                                                     | Subsection #2
+ *      data data data data data    <-- End of area to extract if(includeRegexAfterTable==false)     | Subsection #2
+ *      B                           <-- End of area to extract if(includeRegexAfterTable==true)      | Subsection #2
+ *      data data data data data
+ *
+ *
+ *    In the above example, there is one matching area corresponding to the provided regex table delimiters
+ *    which is composed of two subsections. The above search would be invoked as follows:
+ *    RegexSearch("A",false,"B",false,docName,pageMargins), where pageMargins is a reference to a FilteredArea object
+ *    corresponding to the header and footer margins of the document (this value can be initialized to null).
+ *
+ *    10/29/2017 REM; created.
+ *    1/13/2018  REM; updated detectMatchingAreas to resolve pattern-detection bug
+ *    1/27/2018  REM; added constructors to facilitate header/footer functionality as well as CLI work
+ *    1/30/2018  REM; added static method skeleton for proof-of-concept header work, also added documentation
+ *    2/4/2018   REM; added UpdatesOnResize so that the appropriate data can be passed back to the front-end
+ *    4/14/2018  REM; updated documentation in file to reflect changes in last 2 months...
+ *
+ */
+
+
+public class RegexSearch {
+
+	/*
+	 * This class is leveraged by convertPageSectionToString so that the TextElement ArrayList and the string
+	 * associated with a given PDF page can both be returned to the caller
+	 */
+
+	/* checkSearchesFilterResize
+	 *
+	 * Determines which RegexSearch objects should be modified as a result of the user changing/defining a header filter
+	 * for a page in the document
+	 *
+	 * @param file The PDDocument representation of the file
+	 * @param currentRegexSearches The array of RegexSearch objects used on this document
+	 * @param pageNumOfHeaderResize The number of the page upon which the header filter was resized
+	 * @param pageHeight The height of the page containing the header filter resize event AS IT APPEARED IN THE GUI
+	 * @param newHeaderHeight The new height of the header filter AS IT APPEARED IN THE GUI
+	 * @param previousHeaderHeight The previous height of the header filter AS IT APPEARED IN THE GUI
+	 * @return
+	 */
+	public static ArrayList<UpdatesOnResize> checkSearchesOnFilterResize( PDDocument file,
```

Comment:
Return value is never used. Can we make this method void?

Reply:
Just as a note in case Ross's "rolling back" wasn't sent as a notification itself, this is used on the front end.